core/src/main/java/com/linecorp/armeria/server/management/AppInfoService.java

```diff
@@ -0,0 +1,77 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ */
+
+package com.linecorp.armeria.server.management;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.util.Version;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link HttpService} that provides additional information about configured server.
+ * This functionality is can be used by binding {@link ManagementService}
+ * It provides information about not only deployed application information, which can be specified by user,
+ * but also one about Armeria artifact itself.
+ */
+public enum AppInfoService implements HttpService {
+
+    INSTANCE;
+
+    @Nullable
+    private AppInfo appInfo;
+
+    void setAppInfo(@Nullable AppInfo appInfo) {
+        this.appInfo = appInfo;
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final Version versionInfo = Version.get("armeria", Server.class.getClassLoader());
+
+        final ImmutableMap<String, ImmutableMap<String, String>> appInfoMap;
+        if (appInfo != null) {
+            appInfoMap = ImmutableMap.of(
+                    "app", ImmutableMap.of(
+                            "version", appInfo.version,
+                            "name", appInfo.name,
+                            "description", appInfo.description
+                    )
+            );
+        } else {
+            appInfoMap = ImmutableMap.of();
+        }
+
+        final ImmutableMap<String, ImmutableMap<String, String>> armeriaInfoMap = ImmutableMap.of(
+                "armeria", ImmutableMap.of(
+                        "version", versionInfo.artifactVersion(),
+                        "commit", versionInfo.longCommitHash(),
+                        "repositoryStatus", versionInfo.repositoryStatus()
+                )
+        );
+
+        final ImmutableMap<Object, Object> infoMap = ImmutableMap.builder()
+                                                                 .putAll(appInfoMap)
+                                                                 .putAll(armeriaInfoMap)
+                                                                 .build();
```

Comment:
We don't have to serialize the same value for each request.
Should we cache the value in the member field?

Reply:
Could you guide a bit more about this comment?
What would be the appropriate way to cache the value of serialization result 🤔?

Reply:
```java
public enum AppInfoService implements HttpService {

@Nullablle
private HttpResponse res; // here!

   @Override
    public HttpResponse serve(..) {
      if (res != null) {
         return res;
     }
     ..
    res = HttpResponse.ofJson(infoMap);
    return res;
    }
```
Maybe simply like this?

Reply:
Unfortunately, we can't reuse `HttpResponse` after subscribing to it.
Instead, we can use `AggregatedHttpResponse`. For example:

```java
private static final Version versionInfo = Version.get("armeria", Server.class.getClassLoader());

@Null
private AggregatedHttpResponse aggregatedResponse;

void setAppInfo(@Nullable AppInfo appInfo) {
    byte[] data;
    if (appInfo == null) {
       JacksonUtil.writeValueAsBytes(...);
    }
    aggregatedResponse = AggregatedHttpResponse.of(HttpStatus.OK, MediaType.JSON, data);
}

public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
    assert aggregatedResponse != null;
    return aggregatedResponse.toHttpResponse();
}
```